### PR TITLE
Move response envelope SerDe to its own file

### PR DESCRIPTION
**Pull request by jhendrixMSFT**

### CODE REVIEW

File: packages/autorest.go/src/generator/generator.ts [hunk 1]
```diff
@@ -108,10 +108,17 @@ export async function generateCode(host: AutorestExtensionHost) {
...
       host.writeFile({
         filename: `${filePrefix}responses.go`,
-        content: responses,
+        content: responses.responses,
+        artifactType: 'source-file-go'
+      });
+    }
+    if (responses.serDe.length > 0) {
+      host.writeFile({
+        filename: `${filePrefix}responses_serde.go`,
```
Comment (jhendrixMSFT) [inline packages/autorest.go/src/generator/generator.ts:120]: Technically, there's no "ser" for response envelopes, but `responses_de.go` seems weird.

**Review by chlowell — APPROVED** [review]
What's the motivation for this?

File: packages/autorest.go/src/generator/generator.ts (hunk 1 above)
Comment (chlowell) [reply packages/autorest.go/src/generator/generator.ts:120]: Right, at first glance I'd guess that's German responses

**Comment by jhendrixMSFT** [comment]
We're adding another case where we'll need to generate an unmarshaler on the response envelope. e.g. union types in TSP will likely require an unmarshaler. Cleans up the clutter, especially if we add more cases over time.

**Pull request merged**